contracts/carrot-app/src/handlers/execute.rs

```diff
@@ -181,8 +181,13 @@ fn autocompound(deps: DepsMut, env: Env, info: MessageInfo, app: App) -> AppResu
 
     // If there are external incentives, claim them.
     if !position.claimable_incentives.is_empty() {
+        let asset0_denom = position.asset0.unwrap().denom;
+        let asset1_denom = position.asset1.unwrap().denom;
+
         for coin in try_proto_to_cosmwasm_coins(position.claimable_incentives)? {
-            rewards.add(coin)?;
+            if coin.denom == asset0_denom || coin.denom == asset1_denom {
```

Comment:
Wait, we're not counting other reward types as rewards ? This feels weird

Reply:
We don't autocompound them, because it's not possible to add uosmo to usdc/usdt pool

Reply:
Ah ok we don't swap those tokens yet. Got it. Adding that in v2